TpmTestingPkg/TpmReplayPei/Tool/TpmReplay.py

```diff
@@ -579,6 +592,76 @@ def _add_digest(pcr: int, alg: int, digest: bytes, prehashed: bool = False) -> N
 
     return replay_event_log
 
+def _decode_efi_signature_data_x509(data):
+    cert = x509.load_der_x509_certificate(data)
+    cert_info = {
+        str(cert.subject): {
+            "fingerprint": cert.fingerprint(hashes.SHA1()).hex(":"),
+            "issuer": str(cert.issuer),
+            "not_valid_after_utc": str(cert.not_valid_after_utc),
+        }
+    }
+
+    return cert_info
+
+def _decode_efi_signature_data_sha256(signature) -> dict:
+    return {signature.signature_data.hex(): str(signature.signature_owner)}
+
+def _decode_variable_driver_config(data) -> dict:
+
+    driver_config = {}
+
+    with BytesIO(data) as fs:
+
+        for signature_list in EfiSignatureDatabase(filestream=fs).EslList:
+            if signature_list.signature_data_list is None:  # discard empty EfiSignatureLists
+                continue
+            if signature_list.signature_type == EfiSignatureDataFactory.EFI_CERT_SHA256_GUID:
+                 for signature in signature_list.signature_data_list:
+                    driver_config.update(_decode_efi_signature_data_sha256(signature))
+            elif signature_list.signature_type == EfiSignatureDataFactory.EFI_CERT_X509_GUID:
+                for sig_list in signature_list.signature_data_list:
+                    driver_config.update(
+                        _decode_efi_signature_data_x509(sig_list.signature_data))
+            else:
+                raise Exception("Unsupported signature type %s",
+                                signature_list.signature_type)
+
+    return driver_config
+
+def _decode_variable_authority(data) -> dict:
+    authority_data = {}
+
+    with BytesIO(data) as fs:
+        sig_list = EfiSignatureDataEfiCertX509(
+            decodefs=fs, decodesize=len(data))
+        authority_data.update(
+            _decode_efi_signature_data_x509(sig_list.signature_data))
+
+    return authority_data
+
+def _decode_is_enabled(data) -> dict:
+    return {"enabled": "True" if ord(data) == 1 else "False" }
+
+def _decode_value_dispatch(event_type, name, data) -> dict:
+
+    def _return_nothing(data):
+        return {}
+
+    func = {
+        f"SecureBoot+{EV_EFI_VARIABLE_DRIVER_CONFIG}": _decode_is_enabled,

```

Comment:
This string concat seems unnecessary just for key lookup. I think a tuple would be cleaner, like:

```suggestion
        ("SecureBoot", EV_EFI_VARIABLE_DRIVER_CONFIG): _decode_is_enabled,
```